mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/PrepareGameDialog.kt

```diff
@@ -0,0 +1,68 @@
+package ch.epfl.sdp.mobile.ui.prepare_game
+
+import androidx.compose.animation.animateContentSize
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.zIndex
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.ui.social.Person
+
+@Composable
+fun <P : Person> PrepareGameDialog(
+    state: PrepareGameScreenState<P>,
+    modifier: Modifier = Modifier
+) {
+  val strings = LocalLocalizedStrings.current
+
+  Dialog(
+      modifier = modifier,
+      onCancelClick = { state.onCancelClick() },
+      onConfirmClick = { state.selectedOpponent?.let { state.onPlayClick(it) } },
+      cancelContent = { Text(strings.prepareGameCancel) },
+      confirmContent = { Text(strings.prepareGamePlay) },
+  ) {
+    Column {
+      val lazyListState = rememberLazyListState()
+      val targetElevation =
+          if (lazyListState.firstVisibleItemIndex >= 1 ||
+              lazyListState.firstVisibleItemScrollOffset > 0)
+              4.dp
+          else 0.dp
+      val elevation by animateDpAsState(targetElevation)
+      Surface(modifier = Modifier.zIndex(1f), elevation = elevation) {
```

Comment:
Question : what's the `zIndex` used for ? Do we still need it with our custom `Dialog` layout ?

Reply:
It is used to animate the shadows of the `Dialog` correctly (in the correct order) and is still needed with the custom layout.